### PR TITLE
feat(gatsby-source-contentful): update docs and improve errors

### PR DESCRIPTION
**Pull request by axe312ger**
When working with environments, we need access to master environment as well. This is due to a limitation of the Contentful JS SDK.

Additionally adjusts readme to contain all limitations/restrictions I am aware of.

### CODE REVIEW

File: packages/gatsby-source-contentful/src/__tests__/fetch.js [hunk 1]
```diff
@@ -258,8 +258,13 @@ describe(`Displays troubleshooting tips and detailed plugin options on contentfu
       err.responseData = { status: 404 }
       throw err
     })
+    const masterOptions = { ...options, environment: `master` }
```
Comment (wardpeet) [inline packages/gatsby-source-contentful/src/__tests__/fetch.js:261]: Isn't master the default?

Comment (wardpeet) [reply packages/gatsby-source-contentful/src/__tests__/fetch.js:261]: nvm it's set to env in this test

**Review by wardpeet — DISMISSED** [review]
LGTM!

**Comment by axe312ger** [comment]
closing in favour of https://github.com/gatsbyjs/gatsby/pull/30538 which should run CI

**Pull request closed without merging**